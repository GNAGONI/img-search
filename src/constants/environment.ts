export const PUBLIC_API_KEY = 'abc';
