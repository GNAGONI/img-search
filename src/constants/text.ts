export enum Text { 
  SUCCESS_LOGIN_NOTIFICATION_MESSAGE = "Successfully logged in",
  UNEXPECTED_ERROR_DURING_LOGIN_REQUEST_MESSAGE = "Error during login",
  IMAGES_NOT_FOUND_NOTIFICATION_MESSAGE = "Images not found",
  UNEXPECTED_ERROR_DURING_IMAGE_REQUEST_MESSAGE = "Error during images request",
  NOT_FOUND_ERROR_CODE = "404",
  NOT_FOUND_ERROR_MESSAGE = "The page you’re looking for doesn’t exist.",
  GREETING_MESSAGE = "Hello, ",
}
