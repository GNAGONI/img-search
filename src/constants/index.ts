import { PUBLIC_API_KEY } from "./environment";

export { 
  PUBLIC_API_KEY,
};