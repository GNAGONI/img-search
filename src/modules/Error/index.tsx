import { Typography } from '@mui/material';
import Wrapper from '../../components/Wrapper';
import { Text } from '../../constants';

const Error = () => {
  return (
    <Wrapper orientation="column">
      <Typography variant="h1">
        {Text.NOT_FOUND_ERROR_CODE}
      </Typography>
      <Typography variant="h6">
        {Text.NOT_FOUND_ERROR_MESSAGE}
      </Typography>
    </Wrapper>
  );
};

export default Error;