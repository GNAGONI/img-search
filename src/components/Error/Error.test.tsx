import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Error from './index';
import { Text } from '../../constants';

describe('Error', () => {
  it('should have snapshot', () => {
    const tree = renderer
      .create(<Error />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should behave correctly', () => {
    render(<Error />);
    const errorCode = screen.getByText(Text.NOT_FOUND_ERROR_CODE);
    expect(errorCode).toBeInTheDocument();
    const errorMessage = screen.getByText(Text.NOT_FOUND_ERROR_MESSAGE);
    expect(errorMessage).toBeInTheDocument();
  });
});
