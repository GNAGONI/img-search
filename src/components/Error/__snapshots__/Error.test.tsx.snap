// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should have snapshot 1`] = `
<div
  className="css-d5gdhc"
  orientation="column"
>
  <h1
    className="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
  >
    404
  </h1>
  <h6
    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
  >
    The page you’re looking for doesn’t exist.
  </h6>
</div>
`;
